agriculture_pricing/README.md

```diff
@@ -1,3 +1,154 @@
 
-Example of implementation of agriculture pricing defined [here](https://github.com/Gurobi/modeling-examples/tree/master/agricultural_pricing).
\ No newline at end of file
+Example of implementation of agriculture pricing adapted from [here](https://github.com/Gurobi/modeling-examples/tree/master/agricultural_pricing).
+
+## Data View
+
+Columns marked with __*__ are index columns.
+
+#### `capacity.csv`
+
+| `component`* | `capacity` |
+| --- | --- |
+| __Defines__ set <u>Components</u> ($C$) | __Defines__ param <U>Last Year's Demand |
+
+> Missing index values allowed? : N/A
+
+#### `composition.csv`
+
+| `dairy`* | `component`* | `composition` |
+| --- | --- | -- |
+| Elements __belong__ to <u>Dairy ($D$)</u> | Elements __belong__ to set <u>Components ($C$)</u> | __Defines__ param <u>Composition</u> |
+
+> Missing index values allowed? : No
+
+#### `constants.csv`
+
+| `price_index` |
+| --- |
+| __Defines__ param <u>Price Index</u> |
+
+> Missing index values allowed? : N/A
+
+#### elasticity.csv
+
+| `dairy_1`* | `dairy_2`* | `elasticity` |
+| --- | --- | -- |
+| Elements __belong__ to <u>Dairy ($D$)</u> | Elements __belong__ to <u>Dairy ($D$)</u> | __Defines__ param <u>Elasticity</u> |
+
+> Missing index values allowed? : Yes
+
+#### `market.csv`
+
+| `dairy`* | `demand` | `price` |
+| --- | --- | -- |
+| __Defines__ set <u>Dairy</u> ($D$) | __Defines__ param <u>Last Year's Demand | __Defines__ param <u>Last year's price |
+
+> Missing index values allowed? : N/A
+
+## Model View
+
+
+#### Dairy (Set), $\text{D}$
+
+> Source: From `index columns` of `market.csv`.
+
+#### Components (Set), $\text{C}$
+
+> Source: From `index columns` in `capacity.csv`.
+
+#### Capacity (Param), $\text{capacity}_c$
+
+$$0 \le \text{capacity}_{c} \quad \forall c \in C$$
+
+> Description: Yearly availability of component $c$ (1000 tons).
+
+> Source: From column `capacity` in `capacity.csv`.
+
+#### Composition (Param), $\text{qtyper}_{c,d}$
+
+$$0 \le \text{qtyper}_{c,d}\le 1 \quad \forall c\in C, d\in D$$
+
+> Source: From column `composition` in `composition.csv`
+
+> [!WARNING]
+> How do we specify which column is which in `composition.csv`
+
+> Description: Percentage of component $c$ in dairy product $d$.
+
+#### Last Year's Demand (Param), $\text{consumption}_d$
+
+$$0 \le \text{consumption}_d \quad \forall d \in D$$
+
+> Source: From column `demand` in `market.csv`.
+
+> Description: Last year domestic consumption of dairy product  $d$ (1000 tons).
+
+#### Last Year's Price (Param), $\text{price}_d$
+
+$$0 \le \text{price}_d \quad \forall d \in D$$
+
+> Source : From column `price` in `market.csv`.
+
+> Description: Last year price of dairy product $d$ (dollars/1000 tons).
+
+#### Cross-Elasticities (Param), $\text{elasticity}_{d1,d2}$
+
+$$\text{elasticity}_{d1, d2} \quad \forall d1 \in D, d2 \in D$$
+
+> Default: `0`
+
+> Source: From column `elasticity` in `elasticity.csv`
+
+> [!WARNING]
+> How do we indicate the double index? Do we need to?  
```

Comment:
component_to_sets table is many to one. I suppose it might be possible to not indicate, however, that seems very error-prone because we can't do as many checks